lib/rom/sql/extensions/postgres/types.rb

```diff
@@ -5,21 +5,64 @@ module ROM
   module SQL
     module Types
       module PG
-        Sequel.extension(:pg_json)
+        # UUID
+
+        UUID = Types::Strict::String
+
+        # Array
+
+        Sequel.extension(:pg_array)
+        Sequel.extension(:pg_array_ops)
 
         Array = Dry::Types::Definition
-                .new(Sequel::Postgres::JSONArray)
-                .constructor(Sequel.method(:pg_json))
+                .new(Sequel::Postgres::PGArray)
+
+        def self.Array(db_type)
+          Array.constructor(-> (v) { Sequel.pg_array(v, db_type) }).meta(type: db_type)
+        end
+
+        # JSON
+
+        Sequel.extension(:pg_json)
+        Sequel.extension(:pg_json_ops)
+
+        JSONArray = Dry::Types::Definition
+                    .new(Sequel::Postgres::JSONArray)
+                    .constructor(Sequel.method(:pg_json))
 
-        Hash = Dry::Types::Definition
-               .new(Sequel::Postgres::JSONHash)
-               .constructor(Sequel.method(:pg_json))
+        JSONHash = Dry::Types::Definition
+                   .new(Sequel::Postgres::JSONHash)
+                   .constructor(Sequel.method(:pg_json))
 
-        JSON = Array | Hash
+        JSONOp = Dry::Types::Definition
+                 .new(Sequel::Postgres::JSONOp)
+                 .constructor(Sequel.method(:pg_json))
```

Comment:
See [Sequel's docs](https://github.com/jeremyevans/sequel/blob/40bae5c0426c2a2bf24e92b0a6d880dffb4fbac0/lib/sequel/extensions/pg_json_ops.rb). TLDR Sequel allows to pass various values as a JSON(B) value but they are must be wrapped with `pg_json` call beforehand.